### PR TITLE
Throw exceptions

### PR DESCRIPTION
**Pull request by dominikl**
# What this PR does

Fixes https://github.com/openmicroscopy/openmicroscopy/issues/5033 and adds some missing javadoc.

# Test

There's nothing really to test, code review only; maybe just launch Insight and connect to a server.

### CODE REVIEW

**Comment by jburel** [comment]
travis failure
``
[nomemoryjava] /home/travis/build/openmicroscopy/openmicroscopy/components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java:7347: error: unreported exception Exception; must be caught or declared to be thrown
[nomemoryjava] 	    gw.closeConnector(ctx);
``

File: components/blitz/src/omero/gateway/Gateway.java [hunk 1]
```diff
@@ -1504,7 +1510,7 @@ public void shutDownDerivedConnector(SecurityContext ctx) throws Exception {
         try {
             c.closeDerived(isNetworkUp(true));
         } catch (Throwable e) {
-            new Exception("Cannot close the derived connectors", e);
+            throw new Exception("Cannot close the derived connectors", e);
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/Gateway.java]: while we're here, could we pick a more specific exception for these? (also above)

File: components/blitz/src/omero/gateway/SecurityContext.java [hunk 2]
```diff
@@ -43,8 +43,10 @@
     /** Indicates to generate session for another user. */
     private boolean sudo;
 
+    /** Host/port of the server */
     private ServerInformation serverInformation;
 
+    /** The compression level */
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/SecurityContext.java]: Can we say what the valid range is and what it means? (If not, that's okay, don't spend too much time researching!)

File: components/blitz/src/omero/gateway/Gateway.java (hunk 1 above)
Comment (dominikl) [inline components/blitz/src/omero/gateway/Gateway.java]: I'll remove the whole try catch block, `c.closeDerived(isNetworkUp(true))` actually can't throw an exception anyway.

File: components/blitz/src/omero/gateway/Gateway.java [hunk 3]
```diff
@@ -1292,8 +1296,10 @@ public Connector getConnector(SecurityContext ctx)
      * 
      * @param ctx
      *            The {@link SecurityContext}
+     * @throws Exception
+     *             If the connector is cannot be closed.
      */
-    public void closeConnector(SecurityContext ctx) {
+    public void closeConnector(SecurityContext ctx) throws Exception {
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/Gateway.java]: maybe can throw something narrower

File: components/blitz/src/omero/gateway/SecurityContext.java [hunk 4]
```diff
@@ -138,7 +145,10 @@ public float getCompression() {
 
     /**
      * Set the compression level
-     * @param compression The compression ratio
+     * 
+     * @param compression
+     *            A percentage compression level from 1.00 (100%) to 0.01 (1%)
+     *            (the default is 85%)
      */
     public void setCompression(float compression) {
```
Comment (jburel) [inline components/blitz/src/omero/gateway/SecurityContext.java:153]: Do you want to add a check if greater than 1.00 and less than 0.01 too?

**Comment by jburel** [comment]
Minor comment.
Changes make sense otherwise

**Comment by dominikl** [comment]
*Test* for last commit: Open image in fullviewer, switch between compression levels None, Medium and High, make sure that this still works, i.e. has an effect on the image (None=no artifacts, Medium=a few artifacts, High=Many artifacts).

**Comment by jburel** [comment]
Everything works, thanks for adding the check @dominikl 

Merging

**Pull request merged**